mm2src/adex_cli/src/scenarios/mm2_proc_mng.rs

```diff
@@ -0,0 +1,337 @@
+            TerminateProcess(handy, 1);
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+pub fn start_process(mm2_cfg_file: &Option<String>, coins_file: &Option<String>, log_file: &Option<String>) {
+    let mm2_binary = match get_mm2_binary_path() {
+        Err(_) => return,
+        Ok(path) => path,
+    };
+
+    let Ok(current_dir) = env::current_dir() else {
+	error!("Failet to get current_dir");
+	return
+    };
+
+    let Ok(plist_path)  = get_plist_path() else {return;};
+
+    let plist = format!(
+        r#"<?xml version="1.0" encoding="UTF-8"?>
+        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+        <plist version="1.0">
+        <dict>
+            <key>Label</key>
+            <string>{}</string>
+            <key>ProgramArguments</key>
+            <array>
+                <string>{}</string>
+            </array>
+            <key>WorkingDirectory</key>
+            <string>{}</string>
+            <key>EnvironmentVariables</key>
+            <dict>{}{}{}</dict>
+            <key>RunAtLoad</key>
+            <false/>
+            <key>KeepAlive</key>
+            <false/>
+        </dict>
+        </plist>"#,
+        LAUNCHCTL_MM2_ID,
+        mm2_binary.display(),
+        current_dir.display(),
+        log_file
+            .as_deref()
+            .map(|log_file| format!("<key>MM_LOG</key><string>{log_file}</string>"))
+            .unwrap_or_default(),
+        mm2_cfg_file
+            .as_deref()
+            .map(|cfg_file| format!("<key>MM_CONF_PATH</key><string>{cfg_file}</string>"))
+            .unwrap_or_default(),
+        coins_file
+            .as_deref()
+            .map(|coins_file| format!("<key>MM_COINS_PATH</key><string>{coins_file}</string>"))
+            .unwrap_or_default(),
+    );
+
+    if let Err(error) = fs::write(&plist_path, plist) {
+        error!("Failed to write plist file: {error}");
+        return;
+    }
+
+    match Command::new("launchctl")
+        .arg("enable")
+        .arg(format!("system/{LAUNCHCTL_MM2_ID}").as_str())
+        .spawn()
+    {
+        Ok(_) => debug!("Successfully enabled using launchctl, label: {LAUNCHCTL_MM2_ID}"),
+        Err(error) => error!("Failed to enable process: {error}"),
+    }
+
+    match Command::new("launchctl").arg("load").arg(&plist_path).spawn() {
+        Ok(_) => debug!("Successfully loaded using launchctl, label: {LAUNCHCTL_MM2_ID}"),
+        Err(error) => error!("Failed to load process: {error}"),
+    }
+
+    match Command::new("launchctl").args(["start", LAUNCHCTL_MM2_ID]).spawn() {
+        Ok(_) => info!("Successfully started using launchctl, label: {LAUNCHCTL_MM2_ID}"),
+        Err(error) => error!("Failed to start process: {error}"),
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn get_plist_path() -> Result<PathBuf, ()> {
+    match env::current_dir() {
+        Err(error) => {
+            error!("Failed to get plist_pathcurrent_dir: {error}");
```

Comment:
please add a space in `plist_pathcurrent_dir`

Reply:
done